include/picongpu/initialization/ParserGridDistribution.hpp

```diff
@@ -77,26 +76,39 @@ class ParserGridDistribution
 
     /** Get local Size of this dimension
      *
-     *  \param[in] gpuPos as integer in the range [0, n-1] for this dimension
+     *  \param[in] devicePos as integer in the range [0, n-1] for this dimension
      *  \return uint32_t with local number of cells
      */
     uint32_t
-    getLocalSize( const int gpuPos ) const
+    getLocalSize( const int devicePos ) const
     {
         value_type::const_iterator iter = parsedInput.begin();
-        // go to last gpu of this block b{n}
+        // go to last device of this block b{n}
         int i = iter->second - 1;
 
-        while( i < gpuPos )
+        while( i < devicePos )
         {
             ++iter;
-            // go to last gpu of this block b{n}
+            // go to last device of this block b{n}
             i += iter->second;
         }
 
         return iter->first;
     }
 
+    /** Verify the number of blocks matches the devices
+     *
+     * Check that the number of blocks in a dimension does not exceed the
+     * number of devices for it.
+     *
+     * \param[in] numDevices number of devices for this dimension
+     */
+    void
+    verifyDevices( const int numDevices ) const
+    {
+        PMACC_VERIFY( parsedInput.size() == numDevices );
```

Comment:
The comment says "does not exceed" (and I think that is correct), but the code checks for equality.

Reply:
comment updated

Reply:
I probably get it completely wrong (then just ignore, no need to explain in detail), but why do you check for equality here? [This comment](https://github.com/ComputationalRadiationPhysics/picongpu/blob/7830eafbda291bc7a3d344c0c9dabdd8e27ebd94/include/picongpu/initialization/ParserGridDistribution.hpp#L117-L119) says it fills the `parsedInput` with multipliers. In case there are multipliers greater than 1, would not `parsedInput.size()` be lesser than `numDevices` ?

Reply:
yes, just saw that. `size()` will be too small